File: number/number.py
small = 1
big = 100

def guess_my_number():
    return (small + big) >> 1

def smaller():
    global big
    big = guess_my_number() - 1
    return guess_my_number()

def bigger():
    global small
    small = guess_my_number() + 1
    return guess_my_number()

def start_over():
    global small
    global big
    small = 1
    big = 100
    return guess_my_number()

File: number/number.rb
$small = 1
$big = 100

def guess_my_number
  ($small + $big) << -1
end

def smaller
  $big = guess_my_number - 1
  guess_my_number
end

def bigger
  $small = guess_my_number + 1
  guess_my_number
end

def start_over
  $small = 1
  $big = 100
  guess_my_number
end
